pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +94,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthProvider specifies custom authentication for the kubernetes cluster.
+	AuthProvider *AuthProviderConfig `json:"auth-provider,omitempty"`
```

Comment:
A slice SGTM. It adds ~0 complexity to the single-wrapper use case and future-proofs this a little bit.